diff -rupN d4-7/cmdmain.c d4-hit-miss/cmdmain.c
--- d4-7/cmdmain.c	2013-11-12 14:24:38.297233932 -0600
+++ d4-hit-miss/cmdmain.c	2013-11-12 14:24:27.217233741 -0600
@@ -1130,13 +1130,15 @@ dostats()
 			do1stats (&cc);
 		}
 		else {
-			if (levcache[0][lev] != NULL)
-				do1stats (levcache[0][lev]);
-			if (levcache[1][lev] != NULL)
-				do1stats (levcache[1][lev]);
-			if (levcache[2][lev] != NULL)
-				do1stats (levcache[2][lev]);
-		}
+            if(lev==2){
+				if (levcache[0][lev] != NULL)
+					do1stats (levcache[0][lev]);
+				if (levcache[1][lev] != NULL)
+					do1stats (levcache[1][lev]);
+				if (levcache[2][lev] != NULL)
+					do1stats (levcache[2][lev]);
+				}
+            }
 	}
 }
 
@@ -1170,14 +1172,7 @@ do1stats (d4cache *c)
 	/* Used in bus traffic calculations even if no prefetching. */
 	prefetch_fetch_alltype = 0;
 
-	/*
-	 * Print Header
-	 */
-	printf(	"%s\n", c->name);
-	printf(	" Metrics		      Total	      Instrn	       Data	       Read	      Write	       Misc\n");
-	printf(	" -----------------	      ------	      ------	      ------	      ------	      ------	      ------\n");
-
-	/*
+    /*
 	 * Print Fetch Numbers
 	 */
 	demand_fetch_data = c->fetch[D4XMISC] 
@@ -1186,24 +1181,6 @@ do1stats (d4cache *c)
 	demand_fetch_alltype = demand_fetch_data 
 			  + c->fetch[D4XINSTRN];
 
-	printf(	" Demand Fetches		%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-		demand_fetch_alltype,
-		c->fetch[D4XINSTRN],
-		demand_fetch_data,
-		c->fetch[D4XREAD],
-		c->fetch[D4XWRITE],
-		c->fetch[D4XMISC]);
-
-	floatnum = NONZERO(demand_fetch_alltype);
-
-	printf(	"  Fraction of total	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-		demand_fetch_alltype / floatnum,
-		c->fetch[D4XINSTRN] / floatnum,
-		demand_fetch_data / floatnum,
-		c->fetch[D4XREAD] / floatnum,
-		c->fetch[D4XWRITE] / floatnum,
-		c->fetch[D4XMISC] / floatnum);
-
 	/*
 	 * Prefetching?
 	 */
@@ -1214,48 +1191,8 @@ do1stats (d4cache *c)
 		prefetch_fetch_alltype = prefetch_fetch_data 
 				  + c->fetch[D4PREFETCH+D4XINSTRN];
 
-		printf(	" Prefetch Fetches	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-			prefetch_fetch_alltype,
-			c->fetch[D4PREFETCH+D4XINSTRN],
-			prefetch_fetch_data,
-			c->fetch[D4PREFETCH+D4XREAD],
-			c->fetch[D4PREFETCH+D4XWRITE],
-			c->fetch[D4PREFETCH+D4XMISC]);
-
-		floatnum = NONZERO(prefetch_fetch_alltype);
-
-		printf(	"  Fraction		%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-			prefetch_fetch_alltype / floatnum,
-			c->fetch[D4PREFETCH+D4XINSTRN] / floatnum,
-			prefetch_fetch_data / floatnum,
-			c->fetch[D4PREFETCH+D4XREAD] / floatnum,
-			c->fetch[D4PREFETCH+D4XWRITE] / floatnum,
-			c->fetch[D4PREFETCH+D4XMISC] / floatnum);
-
-		printf(	" Total Fetches		%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-			demand_fetch_alltype + prefetch_fetch_alltype,
-			c->fetch[D4XINSTRN] + c->fetch[D4PREFETCH+D4XINSTRN],
-			demand_fetch_data + prefetch_fetch_data,
-			c->fetch[D4XREAD] + c->fetch[D4PREFETCH+D4XREAD],
-			c->fetch[D4XWRITE] + c->fetch[D4PREFETCH+D4XWRITE],
-			c->fetch[D4XMISC] + c->fetch[D4PREFETCH+D4XMISC]);
-
 		floatnum = NONZERO(demand_fetch_alltype + prefetch_fetch_alltype);
-
-		printf(	"  Fraction		%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-			(demand_fetch_alltype + prefetch_fetch_alltype) / floatnum,
-			(c->fetch[D4XINSTRN] + c->fetch[D4PREFETCH+D4XINSTRN]) / floatnum,
-			(demand_fetch_data + prefetch_fetch_data) / floatnum,
-			(c->fetch[D4XREAD] + c->fetch[D4PREFETCH+D4XREAD]) / floatnum,
-			(c->fetch[D4XWRITE] + c->fetch[D4PREFETCH+D4XWRITE]) / floatnum,
-			(c->fetch[D4XMISC] + c->fetch[D4PREFETCH+D4XMISC]) / floatnum);
-
 	} /* End of prefetching. */
-	printf("\n");
-
-	/*
-	 * End of Fetch Numbers
-	 */
 
 	/*
 	 * Print Miss Numbers
@@ -1266,22 +1203,7 @@ do1stats (d4cache *c)
 	demand_alltype = demand_data 
 			  + c->miss[D4XINSTRN];
 
-	printf(	" Demand Misses		%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-		demand_alltype,
-		c->miss[D4XINSTRN],
-		demand_data,
-		c->miss[D4XREAD],
-		c->miss[D4XWRITE],
-		c->miss[D4XMISC]);
-
-
-	printf(	"  Demand miss rate	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-		demand_alltype / NONZERO(demand_fetch_alltype),
-		c->miss[D4XINSTRN] / NONZERO(c->fetch[D4XINSTRN]),
-		demand_data / NONZERO(demand_fetch_data),
-		c->miss[D4XREAD] / NONZERO(c->fetch[D4XREAD]),
-		c->miss[D4XWRITE] / NONZERO(c->fetch[D4XWRITE]),
-		c->miss[D4XMISC] / NONZERO(c->fetch[D4XMISC]));
+        printf("%12.0f\n", demand_alltype);
 
 	if (c->flags & D4F_CCC) {
 		demand_comp_data = c->comp_miss[D4XMISC] 
@@ -1299,54 +1221,6 @@ do1stats (d4cache *c)
 		      	  + c->conf_miss[D4XWRITE];
 		demand_conf_alltype = demand_conf_data 
 			  + c->conf_miss[D4XINSTRN];
-
-		printf(	"   Compulsory misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-			demand_comp_alltype,
-			c->comp_miss[D4XINSTRN],
-			demand_comp_data,
-			c->comp_miss[D4XREAD],
-			c->comp_miss[D4XWRITE],
-			c->comp_miss[D4XMISC]);
-	    
-		printf(	"   Capacity misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-			demand_cap_alltype,
-			c->cap_miss[D4XINSTRN],
-			demand_cap_data,
-			c->cap_miss[D4XREAD],
-			c->cap_miss[D4XWRITE],
-			c->cap_miss[D4XMISC]);
-	    
-		printf(	"   Conflict misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-			demand_conf_alltype,
-			c->conf_miss[D4XINSTRN],
-			demand_conf_data,
-			c->conf_miss[D4XREAD],
-			c->conf_miss[D4XWRITE],
-			c->conf_miss[D4XMISC]);
-
-		printf(	"   Compulsory fraction	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-			demand_comp_alltype / NONZERO(demand_alltype),
-			c->comp_miss[D4XINSTRN] / NONZERO(c->miss[D4XINSTRN]),
-			demand_comp_data / NONZERO(demand_data),
-			c->comp_miss[D4XREAD] / NONZERO(c->miss[D4XREAD]),
-			c->comp_miss[D4XWRITE] / NONZERO(c->miss[D4XWRITE]),
-			c->comp_miss[D4XMISC] / NONZERO(c->miss[D4XMISC]));
-	    
-		printf(	"   Capacity fraction	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-			demand_cap_alltype / NONZERO(demand_alltype),
-			c->cap_miss[D4XINSTRN]  / NONZERO(c->miss[D4XINSTRN]),
-			demand_cap_data / NONZERO(demand_data),
-			c->cap_miss[D4XREAD] / NONZERO(c->miss[D4XREAD]),
-			c->cap_miss[D4XWRITE] / NONZERO(c->miss[D4XWRITE]),
-			c->cap_miss[D4XMISC] / NONZERO(c->miss[D4XMISC]));
-	    
-		printf(	"   Conflict fraction	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-			demand_conf_alltype / NONZERO(demand_alltype),
-			c->conf_miss[D4XINSTRN] / NONZERO(c->miss[D4XINSTRN]),
-			demand_conf_data / NONZERO(demand_data),
-			c->conf_miss[D4XREAD] / NONZERO(c->miss[D4XREAD]),
-			c->conf_miss[D4XWRITE] / NONZERO(c->miss[D4XWRITE]),
-			c->conf_miss[D4XMISC] / NONZERO(c->miss[D4XMISC]));
 	}
 
 	/*
@@ -1359,22 +1233,6 @@ do1stats (d4cache *c)
 		prefetch_alltype = prefetch_data 
 				  + c->miss[D4PREFETCH+D4XINSTRN];
 
-		printf(	" Prefetch Misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-			prefetch_alltype,
-			c->miss[D4PREFETCH+D4XINSTRN],
-			prefetch_data,
-			c->miss[D4PREFETCH+D4XREAD],
-			c->miss[D4PREFETCH+D4XWRITE],
-			c->miss[D4PREFETCH+D4XMISC]);
-
-		printf(	"  PF miss rate		%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-			prefetch_alltype / NONZERO(prefetch_fetch_alltype),
-			c->miss[D4PREFETCH+D4XINSTRN] / NONZERO(c->fetch[D4PREFETCH+D4XINSTRN]),
-			prefetch_data / NONZERO(prefetch_fetch_data),
-			c->miss[D4PREFETCH+D4XREAD] / NONZERO(c->fetch[D4PREFETCH+D4XREAD]),
-			c->miss[D4PREFETCH+D4XWRITE] / NONZERO(c->fetch[D4PREFETCH+D4XWRITE]),
-			c->miss[D4PREFETCH+D4XMISC] / NONZERO(c->fetch[D4PREFETCH+D4XMISC]));
-
 		if (c->flags & D4F_CCC) {
 			demand_comp_data = c->comp_miss[D4PREFETCH+D4XMISC] 
 			  + c->comp_miss[D4PREFETCH+D4XREAD]
@@ -1392,73 +1250,10 @@ do1stats (d4cache *c)
 			demand_conf_alltype = demand_conf_data 
 			  + c->conf_miss[D4PREFETCH+D4XINSTRN];
 
-			printf(	"   PF compulsory misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-				demand_comp_alltype,
-				c->comp_miss[D4PREFETCH+D4XINSTRN],
-				demand_comp_data,
-				c->comp_miss[D4PREFETCH+D4XREAD],
-				c->comp_miss[D4PREFETCH+D4XWRITE],
-				c->comp_miss[D4PREFETCH+D4XMISC]);
-
-			printf(	"   PF capacity misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-				demand_cap_alltype,
-				c->cap_miss[D4PREFETCH+D4XINSTRN],
-				demand_cap_data,
-				c->cap_miss[D4PREFETCH+D4XREAD],
-				c->cap_miss[D4PREFETCH+D4XWRITE],
-				c->cap_miss[D4PREFETCH+D4XMISC]);
-
-			printf(	"   PF conflict misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-				demand_conf_alltype,
-				c->conf_miss[D4PREFETCH+D4XINSTRN],
-				demand_conf_data,
-				c->conf_miss[D4PREFETCH+D4XREAD],
-				c->conf_miss[D4PREFETCH+D4XWRITE],
-				c->conf_miss[D4PREFETCH+D4XMISC]);
-
-			printf(	"   PF compulsory fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-				demand_comp_alltype / NONZERO(prefetch_alltype),
-				c->comp_miss[D4PREFETCH+D4XINSTRN] / NONZERO(c->miss[D4PREFETCH+D4XINSTRN]),
-				demand_comp_data / NONZERO(prefetch_data),
-				c->comp_miss[D4PREFETCH+D4XREAD] / NONZERO(c->miss[D4PREFETCH+D4XREAD]),
-				c->comp_miss[D4PREFETCH+D4XWRITE] / NONZERO(c->miss[D4PREFETCH+D4XWRITE]),
-				c->comp_miss[D4PREFETCH+D4XMISC] / NONZERO(c->miss[D4PREFETCH+D4XMISC]));
-
-			printf(	"   PF capacity fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-				demand_cap_alltype / NONZERO(prefetch_alltype),
-				c->cap_miss[D4PREFETCH+D4XINSTRN] / NONZERO(c->miss[D4PREFETCH+D4XINSTRN]),
-				demand_cap_data / NONZERO(prefetch_data),
-				c->cap_miss[D4PREFETCH+D4XREAD] / NONZERO(c->miss[D4PREFETCH+D4XREAD]),
-				c->cap_miss[D4PREFETCH+D4XWRITE] / NONZERO(c->miss[D4PREFETCH+D4XWRITE]),
-				c->cap_miss[D4PREFETCH+D4XMISC] / NONZERO(c->miss[D4PREFETCH+D4XMISC]));
-
-			printf(	"   PF conflict fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-				demand_conf_alltype / NONZERO(prefetch_alltype),
-				c->conf_miss[D4PREFETCH+D4XINSTRN] / NONZERO(c->miss[D4PREFETCH+D4XINSTRN]),
-				demand_conf_data / NONZERO(prefetch_data),
-				c->conf_miss[D4PREFETCH+D4XREAD] / NONZERO(c->miss[D4PREFETCH+D4XREAD]),
-				c->conf_miss[D4PREFETCH+D4XWRITE] / NONZERO(c->miss[D4PREFETCH+D4XWRITE]),
-				c->conf_miss[D4PREFETCH+D4XMISC] / NONZERO(c->miss[D4PREFETCH+D4XMISC]));
-		}
-
-		printf(	" Total Misses		%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-			demand_alltype + prefetch_alltype,
-			c->miss[D4XINSTRN] + c->miss[D4PREFETCH+D4XINSTRN],
-			demand_data + prefetch_data,
-			c->miss[D4XREAD] + c->miss[D4PREFETCH+D4XREAD],
-			c->miss[D4XWRITE] + c->miss[D4PREFETCH+D4XWRITE],
-			c->miss[D4XMISC] + c->miss[D4PREFETCH+D4XMISC]);
-
-		printf(	"  Total miss rate	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-			(demand_alltype + prefetch_alltype) / NONZERO(demand_fetch_alltype + prefetch_fetch_alltype),
-			(c->miss[D4XINSTRN] + c->miss[D4PREFETCH+D4XINSTRN]) / NONZERO(c->fetch[D4XINSTRN] + c->fetch[D4PREFETCH+D4XINSTRN]),
-			(demand_data + prefetch_data) / NONZERO(demand_fetch_data + prefetch_fetch_data),
-			(c->miss[D4XREAD] + c->miss[D4PREFETCH+D4XREAD]) / NONZERO(c->fetch[D4XREAD] + c->fetch[D4PREFETCH+D4XREAD]),
-			(c->miss[D4XWRITE] + c->miss[D4PREFETCH+D4XWRITE]) / NONZERO(c->fetch[D4XWRITE] + c->fetch[D4PREFETCH+D4XWRITE]),
-			(c->miss[D4XMISC] + c->miss[D4PREFETCH+D4XMISC]) / NONZERO(c->fetch[D4XMISC] + c->fetch[D4PREFETCH+D4XMISC]));
+		}
 
 	} /* End of prefetch misses. */
-	printf("\n");
+//	printf("\n");
 
 	/*
 	 * End of Misses Numbers
@@ -1475,22 +1270,6 @@ do1stats (d4cache *c)
 		demand_alltype = demand_data 
 			  + c->blockmiss[D4XINSTRN];
 
-		printf(	" Demand Block Misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-			demand_alltype,
-			c->blockmiss[D4XINSTRN],
-			demand_data,
-			c->blockmiss[D4XREAD],
-			c->blockmiss[D4XWRITE],
-			c->blockmiss[D4XMISC]);
-
-		printf(	"  DB miss rate		%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-			demand_alltype / NONZERO(demand_fetch_alltype),
-			c->blockmiss[D4XINSTRN] / NONZERO(c->fetch[D4XINSTRN]),
-			demand_data / NONZERO(demand_fetch_data),
-			c->blockmiss[D4XREAD] / NONZERO(c->fetch[D4XREAD]),
-			c->blockmiss[D4XWRITE] / NONZERO(c->fetch[D4XWRITE]),
-			c->blockmiss[D4XMISC] / NONZERO(c->fetch[D4XMISC]));
-
 		if (c->flags & D4F_CCC) {
 			demand_comp_data = c->comp_blockmiss[D4XMISC] 
 			  + c->comp_blockmiss[D4XREAD]
@@ -1507,55 +1286,8 @@ do1stats (d4cache *c)
 		      	  + c->conf_blockmiss[D4XWRITE];
 			demand_conf_alltype = demand_conf_data 
 			  + c->conf_blockmiss[D4XINSTRN];
-
-			printf(	"   DB compulsory misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-				demand_comp_alltype,
-				c->comp_blockmiss[D4XINSTRN],
-				demand_comp_data,
-				c->comp_blockmiss[D4XREAD],
-				c->comp_blockmiss[D4XWRITE],
-				c->comp_blockmiss[D4XMISC]);
-
-			printf(	"   DB capacity misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-				demand_cap_alltype,
-				c->cap_blockmiss[D4XINSTRN],
-				demand_cap_data,
-				c->cap_blockmiss[D4XREAD],
-				c->cap_blockmiss[D4XWRITE],
-				c->cap_blockmiss[D4XMISC]);
-
-			printf(	"   DB conflict misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-				demand_conf_alltype,
-				c->conf_blockmiss[D4XINSTRN],
-				demand_conf_data,
-				c->conf_blockmiss[D4XREAD],
-				c->conf_blockmiss[D4XWRITE],
-				c->conf_blockmiss[D4XMISC]);
-
-			printf(	"   DB compulsory fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-				demand_comp_alltype / NONZERO(demand_alltype),
-				c->comp_blockmiss[D4XINSTRN] / NONZERO(c->blockmiss[D4XINSTRN]),
-				demand_comp_data / NONZERO(demand_data),
-				c->comp_blockmiss[D4XREAD] / NONZERO(c->blockmiss[D4XREAD]),
-				c->comp_blockmiss[D4XWRITE] / NONZERO(c->blockmiss[D4XWRITE]),
-				c->comp_blockmiss[D4XMISC]) / NONZERO(c->blockmiss[D4XMISC]);
-
-			printf(	"   DB capacity fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-				demand_cap_alltype / NONZERO(demand_alltype),
-				c->cap_blockmiss[D4XINSTRN] / NONZERO(c->blockmiss[D4XINSTRN]),
-				demand_cap_data / NONZERO(demand_data),
-				c->cap_blockmiss[D4XREAD] / NONZERO(c->blockmiss[D4XREAD]),
-				c->cap_blockmiss[D4XWRITE] / NONZERO(c->blockmiss[D4XWRITE]),
-				c->cap_blockmiss[D4XMISC] / NONZERO(c->blockmiss[D4XMISC]));
-	    
-			printf(	"   DB conflict fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-				demand_conf_alltype / NONZERO(demand_alltype),
-				c->conf_blockmiss[D4XINSTRN] / NONZERO(c->blockmiss[D4XINSTRN]),
-				demand_conf_data / NONZERO(demand_data),
-				c->conf_blockmiss[D4XREAD] / NONZERO(c->blockmiss[D4XREAD]),
-				c->conf_blockmiss[D4XWRITE] / NONZERO(c->blockmiss[D4XWRITE]),
-				c->conf_blockmiss[D4XMISC] / NONZERO(c->blockmiss[D4XMISC]));
-		}
+			}
+    }
 
 		/*
 		 * Prefetch block misses?
@@ -1567,22 +1299,6 @@ do1stats (d4cache *c)
 			prefetch_alltype = prefetch_data 
 				  + c->blockmiss[D4PREFETCH+D4XINSTRN];
 
-			printf(	" Prefetch Block Misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-				prefetch_alltype,
-				c->blockmiss[D4PREFETCH+D4XINSTRN],
-				prefetch_data,
-				c->blockmiss[D4PREFETCH+D4XREAD],
-				c->blockmiss[D4PREFETCH+D4XWRITE],
-				c->blockmiss[D4PREFETCH+D4XMISC]);
-
-			printf(	"  PFB miss rate		%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-				prefetch_alltype / NONZERO(prefetch_fetch_alltype),
-				c->blockmiss[D4PREFETCH+D4XINSTRN] / NONZERO(c->fetch[D4PREFETCH+D4XINSTRN]),
-				prefetch_data / NONZERO(prefetch_fetch_data),
-				c->blockmiss[D4PREFETCH+D4XREAD] / NONZERO(c->fetch[D4PREFETCH+D4XREAD]),
-				c->blockmiss[D4PREFETCH+D4XWRITE] / NONZERO(c->fetch[D4PREFETCH+D4XWRITE]),
-				c->blockmiss[D4PREFETCH+D4XMISC] / NONZERO(c->fetch[D4PREFETCH+D4XMISC]));
-
 			if (c->flags & D4F_CCC) {
 				demand_comp_data = c->comp_blockmiss[D4PREFETCH+D4XMISC] 
 					+ c->comp_blockmiss[D4PREFETCH+D4XREAD]
@@ -1599,74 +1315,10 @@ do1stats (d4cache *c)
 					+ c->conf_blockmiss[D4PREFETCH+D4XWRITE];
 				demand_conf_alltype = demand_conf_data 
 					+ c->conf_blockmiss[D4PREFETCH+D4XINSTRN];
-
-				printf(	"   PFB comp misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-					demand_comp_alltype,
-					c->comp_blockmiss[D4PREFETCH+D4XINSTRN],
-					demand_comp_data,
-					c->comp_blockmiss[D4PREFETCH+D4XREAD],
-					c->comp_blockmiss[D4PREFETCH+D4XWRITE],
-					c->comp_blockmiss[D4PREFETCH+D4XMISC]);
-
-				printf(	"   PFB cap misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-					demand_cap_alltype,
-					c->cap_blockmiss[D4PREFETCH+D4XINSTRN],
-					demand_cap_data,
-					c->cap_blockmiss[D4PREFETCH+D4XREAD],
-					c->cap_blockmiss[D4PREFETCH+D4XWRITE],
-					c->cap_blockmiss[D4PREFETCH+D4XMISC]);
-
-				printf(	"   PFB conf misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-					demand_conf_alltype,
-					c->conf_blockmiss[D4PREFETCH+D4XINSTRN],
-					demand_conf_data,
-					c->conf_blockmiss[D4PREFETCH+D4XREAD],
-					c->conf_blockmiss[D4PREFETCH+D4XWRITE],
-					c->conf_blockmiss[D4PREFETCH+D4XMISC]);
-
-				printf(	"   PFB comp fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-					demand_comp_alltype / NONZERO(prefetch_alltype),
-					c->comp_blockmiss[D4PREFETCH+D4XINSTRN] / NONZERO(c->blockmiss[D4PREFETCH+D4XINSTRN]),
-					demand_comp_data / NONZERO(prefetch_data),
-					c->comp_blockmiss[D4PREFETCH+D4XREAD] / NONZERO(c->blockmiss[D4PREFETCH+D4XREAD]),
-					c->comp_blockmiss[D4PREFETCH+D4XWRITE] / NONZERO(c->blockmiss[D4PREFETCH+D4XWRITE]),
-					c->comp_blockmiss[D4PREFETCH+D4XMISC] / NONZERO(c->blockmiss[D4PREFETCH+D4XMISC]));
-	    
-				printf(	"   PFB cap fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-					demand_cap_alltype / NONZERO(prefetch_alltype),
-					c->cap_blockmiss[D4PREFETCH+D4XINSTRN] / NONZERO(c->blockmiss[D4PREFETCH+D4XINSTRN]),
-					demand_cap_data / NONZERO(prefetch_data),
-					c->cap_blockmiss[D4PREFETCH+D4XREAD] / NONZERO(c->blockmiss[D4PREFETCH+D4XREAD]),
-					c->cap_blockmiss[D4PREFETCH+D4XWRITE] / NONZERO(c->blockmiss[D4PREFETCH+D4XWRITE]),
-					c->cap_blockmiss[D4PREFETCH+D4XMISC] / NONZERO(c->blockmiss[D4PREFETCH+D4XMISC]));
-
-	    			printf(	"   PFB conf fract	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-					demand_conf_alltype / NONZERO(prefetch_alltype),
-					c->conf_blockmiss[D4PREFETCH+D4XINSTRN] / NONZERO(c->blockmiss[D4PREFETCH+D4XINSTRN]),
-					demand_conf_data / NONZERO(prefetch_data),
-					c->conf_blockmiss[D4PREFETCH+D4XREAD] / NONZERO(c->blockmiss[D4PREFETCH+D4XREAD]),
-					c->conf_blockmiss[D4PREFETCH+D4XWRITE] / NONZERO(c->blockmiss[D4PREFETCH+D4XWRITE]),
-					c->conf_blockmiss[D4PREFETCH+D4XMISC] / NONZERO(c->blockmiss[D4PREFETCH+D4XMISC]));
 			}
-
-			printf(	" Total Block Misses	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f	%12.0f\n",
-				demand_alltype + prefetch_alltype,
-				c->blockmiss[D4XINSTRN] + c->blockmiss[D4PREFETCH+D4XINSTRN],
-				demand_data + prefetch_data,
-				c->blockmiss[D4XREAD] + c->blockmiss[D4PREFETCH+D4XREAD],
-				c->blockmiss[D4XWRITE] + c->blockmiss[D4PREFETCH+D4XWRITE],
-				c->blockmiss[D4XMISC] + c->blockmiss[D4PREFETCH+D4XMISC]);
-
-			printf(	"  Tot blk miss rate	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f	%12.4f\n",
-				(demand_alltype + prefetch_alltype) / NONZERO(demand_fetch_alltype + prefetch_fetch_alltype),
-				(c->blockmiss[D4XINSTRN] + c->blockmiss[D4PREFETCH+D4XINSTRN]) / NONZERO(c->fetch[D4XINSTRN] + c->fetch[D4PREFETCH+D4XINSTRN]),
-				(demand_data + prefetch_data) / NONZERO(demand_fetch_data + prefetch_fetch_data),
-				(c->blockmiss[D4XREAD] + c->blockmiss[D4PREFETCH+D4XREAD]) / NONZERO(c->fetch[D4XREAD] + c->fetch[D4PREFETCH+D4XREAD]),
-				(c->blockmiss[D4XWRITE] + c->blockmiss[D4PREFETCH+D4XWRITE]) / NONZERO(c->fetch[D4XWRITE] + c->fetch[D4PREFETCH+D4XWRITE]),
-				(c->blockmiss[D4XMISC] + c->blockmiss[D4PREFETCH+D4XMISC]) / NONZERO(c->fetch[D4XMISC] + c->fetch[D4PREFETCH+D4XMISC]));
 		} /* End of prefetch block misses */
-		printf("\n");
-	} /* End of block misses */
+//		printf("\n");
+//	} /* End of block misses */
 	/*
 	 * End of Block Misses Numbers
 	 */
@@ -1674,7 +1326,7 @@ do1stats (d4cache *c)
 	/*
 	 * Report multiblock and traffic to/from memory
 	 */
-	printf( " Multi-block refs      %12.0f\n",
+/*	printf( " Multi-block refs      %12.0f\n",
 		c->multiblock);
 	printf(	" Bytes From Memory	%12.0f\n",
 		c->bytes_read);
@@ -1686,9 +1338,12 @@ do1stats (d4cache *c)
 		c->bytes_written / NONZERO(c->fetch[D4XWRITE]));
 	printf(	" Total Bytes r/w Mem	%12.0f\n",
 		c->bytes_read + c->bytes_written);
+                
 	printf(	" ( / Demand Fetches)	%12.4f\n",
-		(c->bytes_read + c->bytes_written) / NONZERO(demand_fetch_alltype));
+		(c->bytes_read + c->bytes_written) / NONZERO(demand_fetch_alltype)); 
+                
 	printf("\n");
+*/
 }
 
 #undef NONZERO
@@ -1958,7 +1613,7 @@ main (int argc, char **argv)
 	if (cd == NULL)
 		cd = ci;	/* for unified L1 cache */
 
-	printf ("---Dinero IV cache simulator, version %s\n", D4VERSION);
+/*	printf ("---Dinero IV cache simulator, version %s\n", D4VERSION);
 	printf ("---Written by Jan Edler and Mark D. Hill\n");
 	printf ("---Copyright (C) 1997 NEC Research Institute, Inc. and Mark D. Hill.\n");
 	printf ("---All rights reserved.\n");
@@ -1966,8 +1621,8 @@ main (int argc, char **argv)
 	printf ("---See -copyright option for details\n");
 
 	summarize_caches (ci, cd);
-
-	printf ("\n---Simulation begins.\n");
+*/
+//	printf ("\n---Simulation begins.\n");
 	tintcount = stat_interval;
 	flcount = flushcount;
 	while (1) {
@@ -1984,8 +1639,9 @@ main (int argc, char **argv)
 		default:	  d4ref (cd, r);  break;
 		}
 		tmaxcount += 1;
+		dostats();
 		if (tintcount > 0 && (tintcount -= 1) <= 0) {
-			dostats();
+//			dostats();
 			tintcount = stat_interval;
 		}
 		if (flcount > 0 && (flcount -= 1) <= 0) {
@@ -2007,8 +1663,8 @@ done:
 	r.address = 0;
 	r.size = 0;
 	d4ref (cd, r);
-	printf ("---Simulation complete.\n");
+//	printf ("---Simulation complete.\n");
 	dostats();
-	printf ("---Execution complete.\n");
+//	printf ("---Execution complete.\n");
 	return 0;
 }
